File: src/python/wreck/text_tools.py
def face_message(face_order, f_code_2_name=None):
    if f_code_2_name is None:
        f_code_2_name = {
            'n': 'back',
            's': 'front',
            'e': 'right',
            'w': 'left',
            't': 'top',
            'b': 'bottom'
        }

    if len(face_order) == 1:
        face_msg = '{} face'.format(f_code_2_name[face_order[0]])
    else:
        face_msg = ', '.join(map(f_code_2_name.get, face_order[:-1])) + ' and {} faces'.format(f_code_2_name[face_order[-1]])

    return face_msg


def uncapitalize(msg):
    return msg[0].lower() + msg[1:]

File: old/boxes/text_tools.ts
import {Face} from './datatypes';

export function face_message(face_order: Face[], f_code_2_name?: Map<Face, string>){
    if (f_code_2_name === undefined) {
        f_code_2_name = new Map<Face, string>([
            [Face.n, 'back'],
            [Face.s, 'front'],
            [Face.e, 'right'],
            [Face.w, 'left'],
            [Face.t, 'top'],
            [Face.b, 'bottom']
        ]);
    }

    if (face_order.length == 1) {
        return f_code_2_name.get(face_order[0]) + ' face';
    } else {
        return face_order.slice(0, -1).map((x) => f_code_2_name.get(x)).join(', ') + ' and ' + f_code_2_name.get(face_order[face_order.length - 1]) + ' faces';
    }
}
